src/networks.c

```diff
@@ -582,7 +500,7 @@ Warning: This function does not check whether the given origin and class
 fall into the current batch.  Run checkifcurrent to check that if you want.
 */
 int nodeclass2origin(network_type *network, int originNode, int class) {
-    return (class * network->numZones + originNode);
+    return (class * network->numZones + originNode) % network->batchSize;

```

Comment:
We modified this and removed the modulo because this assumes demand matrix is currently batchSize, but during the reading of the network at the start on Line 280-290 of fileio.c, it will overwrite the demand matrix for just the first class